lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -123,10 +123,13 @@ function isObjectLocked(bucket, objectMD, headers) {
         if (objectLegalHold) {
             return true;
         }
-        const objectRetention = objectMD.retentionInfo;
+        let objectRetention = objectMD.retentionInfo;
```

Comment:
I see a lot of objectMD usage where properties are accessed directly. The reason we have ObjectMD is so that we avoid this and set/get properties using setters/getters. It is also the root cause of this bug.

Reply:
As discussed, how we use objectMD is flawed throughout Cloudserver. It would take a good bit of refactoring to make using getters/setters viable, which is outside the scope of this pr

Reply:
Thanks for looking into it @dora-korpar . Let's follow up as a tech debt to address.